ros2_ws/Dockerfile

```diff
@@ -56,6 +56,22 @@ RUN ( \
   ) > /etc/ssh/sshd_config_development \
   && mkdir /run/sshd
 
+FROM base-dependencies AS build-go
+ARG TARGETPLATFORM
+ARG GOVERSION=1.24.2
+
+RUN wget https://go.dev/dl/go${GOVERSION}.linux-${TARGETPLATFORM#*/}.tar.gz -O /tmp/go.tar.gz && \
+  sudo tar -C /usr/local -xzf /tmp/go.tar.gz && \
+  rm /tmp/go.tar.gz
+ENV PATH="${PATH}:/usr/local/go/bin"
+WORKDIR /go
+COPY config/validate_json.go .
+RUN --mount=type=cache,target=/go/pkg/mod \
+  go mod init validate_json.go && go mod tidy
```

Comment:
Shouldn't run `tidy` in the Docker image as it might mutate the `go.mod`/`go.sum` and thus install things that aren't pinned in source. Instead we should have the `go.mod` in source control.
Maybe just create a `validate_json` subfolder in `config` so you can group all the files easily? (you can then also avoid the `-o validate_json` as `go build` derive the output name from the name of the parent folder of the `package main`, another funky quirk).